lib/koans/10_processes.ex

```diff
@@ -26,15 +45,71 @@ defmodule Processes do
   end
 
   koan "A common pattern is to include the sender in the message, so that it can reply" do
-    pid = spawn(fn -> receive do
-                         {:hello, sender} -> send sender, :how_are_you?
-                      end
-                 end)
+    greeter = fn ->
+      receive do
+        {:hello, sender} -> send sender, :how_are_you?
+      end
+    end
+
+    pid = spawn(greeter)
 
     send pid, {:hello, self()}
     assert_receive ___
   end
 
+  def yelling_echo_loop do
+    receive do
+      {caller, value} ->
+        send caller, String.upcase(value)
+        yelling_echo_loop
+    end
+  end
+
+  koan "Use tail recursion (calling a function as the very last statement) to receive multiple messages" do
```

Comment:
I'm sort of on the fence about it. On one hand, the example certainly works without the explanation. On the other hand, it's extremely important to this style of code.

That said, it doesn't _actually_ explain tail recursion, it just throws the term out there. Perhaps a better way to revealing this would be to have this example (without mention of tail recursion) and then another example that illustrates why tail recursion is important... Not sure what that would look like.

Thoughts?